### PR TITLE
PWX-37733 Add pod name in system metadata device for dmthin on ocp facd

### PR DESCRIPTION
**Pull request by ssz1997**
**Which issue(s) this PR fixes** (optional)
https://purestorage.atlassian.net/browse/PWX-37733

**Special notes for your reviewer**:
Along with https://github.com/pure-px/porx/pull/13643 I successfully run a px cluster with dmthin with ocp facd

### CODE REVIEW

File: drivers/storage/portworx/preflight.go [hunk 1]
```diff
@@ -410,7 +410,19 @@ func (u *preFlightPortworx) processPassedChecks(recorder record.EventRecorder) {
...
 		} else if pxutil.IsPure(u.cluster) {
-			cmetaData = DefCmetaFACD
+			var podName string
+		specsItr:
+			for _, spec := range *u.cluster.Spec.CloudStorage.DeviceSpecs {
+				options := strings.Split(spec, ",")
+				for _, option := range options {
+					keyValPair := strings.Split(option, "=")
+					if len(keyValPair) == 2 && keyValPair[0] == "pod" {
+						podName = option
```
Comment (pp511) [inline drivers/storage/portworx/preflight.go:420]: Let's not name the variable name as podName here. It appears as if it's the name of the pod while it actually is the key, value pair. Maybe let's just suffix it with "Option".

Comment (jrivera-px) [reply drivers/storage/portworx/preflight.go:420]: We actually have a drive format validation check in px-runc pre-flight.   So we need to make sure that the check in px-runc will accept this format `"size=49,pod=<name>"`

sudo /opt/pwx/bin/px-runc pre-flight -c .... -s size=200,pod=test ... -T px-storeV2 ... --cloud_provider pure ...

Comment (ssz1997) [reply drivers/storage/portworx/preflight.go:420]: People did some changes in `px-runc.go` that will be cherry-picked into the px release branch, and I believe this commit serves the purpose of skipping the validation: https://github.com/pure-px/porx/commit/e1d44d49737daf6f11c083a5c228a74c58913579

Comment (ssz1997) [reply drivers/storage/portworx/preflight.go:420]: I also cherry-picked this commit to my local test and px didn't complain with the format

**Review by pp511 — APPROVED** [review]
lgtm

**Pull request merged**